compiler/src/iree/compiler/Codegen/Sandbox/LinalgTensorCodegenDriver.cpp

```diff
@@ -366,6 +411,128 @@ void LinalgFusePass::runOnOperation() {
     return signalPassFailure();
   }
 }
+namespace {
+/// Pattern to wrap splitReduction transformation. This changes the
+/// lowering_config for the split op and keeps the same lowering_config for the
+/// reduction op.
+struct CodegenSplitReduction
+    : public OpInterfaceRewritePattern<linalg::LinalgOp> {
+  CodegenSplitReduction(MLIRContext *context, int64_t size,
+                        uint64_t unrollFactor,
+                        linalg::LinalgTransformationFilter filter,
+                        PatternBenefit benefit = 1)
+      : OpInterfaceRewritePattern<linalg::LinalgOp>(context, benefit),
+        size(size),
+        unrollFactor(unrollFactor),
+        filter(std::move(filter)) {}
+
+  LogicalResult matchAndRewrite(linalg::LinalgOp op,
+                                PatternRewriter &rewriter) const override {
+    // Make sure that only generic ops are supported, only one input and the
+    // innermost tensor dimension is reduction dimension.  Everything else is
+    // copied from the original SplitReduction Transformation.
+    if (failed(filter.checkAndNotify(rewriter, op)) ||
+        !op.hasTensorSemantics() || op.getNumReductionLoops() != 1 ||
+        op.getNumOutputs() != 1 || !op.hasOnlyProjectedPermutations() ||
+        !isa<linalg::GenericOp>(op) || op.getNumInputs() != 1)
+      return rewriter.notifyMatchFailure(op, "precondition not met");
+    SmallVector<unsigned> dims;
+    op.getReductionDims(dims);
+    AffineMap map = op.getTiedIndexingMap(op.getInputOperand(0));
+    unsigned lastIdx = map.getNumResults() - 1;
+    unsigned lastDim = map.getDimPosition(lastIdx);
+    if (lastDim != dims[0])
+      return rewriter.notifyMatchFailure(op, "precondition not met");
+
+    linalg::ControlSplitReductionFn fn = [this, lastIdx](linalg::LinalgOp op) {
+      return linalg::SplitReductionOptions{size, lastIdx, true};
+    };
+
+    auto opNumLoops = op.getNumLoops();
+
+    // 1) Tile to extract a single vector-length array.
+    SmallVector<int64_t> tileSizesSVFirst(opNumLoops, 1);
+    tileSizesSVFirst[opNumLoops - 1] = 0;
+    ArrayRef<int64_t> tileSizesFirst(tileSizesSVFirst);
+    linalg::LinalgTilingOptions optionsFirst;
+    optionsFirst.setTileSizes(tileSizesFirst);
```

Comment:
We don't have to declare the ArrayRef type. The `SmallVector` will be casted to ArrayRef type in this case. Also, we can make setting options cleaner. I think the below snippet work, can you give it a shot?

```suggestion
    SmallVector<int64_t> tileSizesSVFirst(opNumLoops, 1);
    tileSizesSVFirst[opNumLoops - 1] = 0;
    auto optionsFirst = linalg::LinalgTilingOptions().setTileSizes(tileSizesSVFirst);
```